lib/settings.js

```diff
@@ -96,11 +96,14 @@ class Settings extends EventEmitter {
       this._defaultSettingsFilePath;
   }
 
-  _handleObjectChanged() {
+  _handleObjectChanged(keyPath = '') {
     const settingsFilePath = this._getSettingsFilePath();
     jsonfile.writeFileSync(settingsFilePath, this._cacheObj);
-    _.forEach(this._watchHandlers, handler => {
-      handler.populate();
+    const relatedHandler = keyPath ?
```

Comment:
We don't need this as long as we make sure keyPath is a string (are we doing handler._keyPath or handler.keyPath?

```
const path = keypath || ''
_(this._watchHandlers)
  .filter(h => h._keyPath.startsWith(keyPath))
  .each(h => h.checkChanged)
```

Reply:
there is a case that we need to inform for all handers such as (setPath => change config file location)

Reply:
i call this method to write the whole current object into new setting file (the same behavior of electron-settings). => i think we should notify all observers again for sure. Actually our Crypto Miner does not use this api

Reply:
Changing file path should definitely not trigger value change function since the values don't change. (And the setPath should actually write the existing object there)